snooty/parser.py

```diff
@@ -591,29 +593,39 @@ def handle_directive(
             openapi_fileid, filepath = util.reroot_path(
                 FileId(argument_text), self.docpath, self.project_config.source_path
             )
+            swagger_parse = True if "swagger-parse" in options else False
 
             try:
-                with open(filepath) as f:
-                    openapi = OpenAPI.load(f)
+                # swagger-parse leaves parsing to the Swagger React library in the frontend
+                if swagger_parse:
+                    with open(filepath) as f:
+                        spec = json.dumps(safe_load(f))
+
+                        span = (line,)
+                        openapi_spec = n.OpenAPISpec(span, spec)
+                        doc.children.append(openapi_spec)
```

Comment:
I'm a little hesitant as to whether the file contents should be appended to the node's `children` property. Thus far, `children` has always represented a series of AST nodes, never a string like this. Would love to hear from @i80and!

Reply:
This part was inspired by the way `literalinclude` was parsed in that it appends a `code` node as its only child, with the contents of the file added as that node's `value`.